### PR TITLE
Rate limiter

### LINKED ISSUE #12: Add simple HTTP rate limit
Let's protect our API/HTTP with a simple rate limiting mechanism.

- We can write it in pure Go using channels (see [example](https://gobyexample.com/rate-limiting)).
- Let's implement limiting based on remote IPs (just like [Rails](https://github.com/rails/rails/pull/50490/files#diff-aefcfa4f79f572a043b109d61c7d60feddc64d05e7c5e34abeaa4721c9ab60b9))
- Limiting options (interval, etc.) must be configurable

### PR DESCRIPTION
**Pull request by prog-supdex**
Resolve #12

### CODE REVIEW

File: pkg/cli/options.go [hunk 1]
```diff
@@ -38,6 +38,22 @@ func NewCliConfig() (cfg *config.Config, err error, stopProgram bool) {
 				Category:    "Server:",
 				Destination: &cfg.Server.ServerPort,
 			},
+			&cli.IntFlag{
+				Name:        "rate_limit",
+				Usage:       "how many request can perform during the window of time",
+				Value:       2,
```
Comment (palkan) [inline pkg/cli/options.go]: 2 is too low for the default; I would go with the default 0 (meaning no limit) or smth reasonable, say, 100 (and 0 must be treated as no limit anyway)

File: pkg/server/config.go [hunk 2]
```diff
@@ -2,6 +2,12 @@ package server
 
 type Config struct {
 	ServerPort int
+	Request    ConfigRequest
+}
+
+type ConfigRequest struct {
```
Comment (palkan) [inline pkg/server/config.go]: No need to split the Config struct; rate limiting is a server feature

File: pkg/cli/options.go (hunk 1 above)
Comment (prog-supdex) [reply pkg/cli/options.go]: thanks!
yea, it was just a temporary param, but I fixed it and added 0 value as unlimited

File: pkg/server/config.go (hunk 2 above)
Comment (prog-supdex) [reply pkg/server/config.go]: done

**Pull request merged**